pkg/common/kv.go

```diff
@@ -36,14 +36,25 @@ func (s *SimpleKV) GetValue() interface{} {
 	return s.Value
 }
 
-// KVs will store SimpleKV collection as map
-type KVs struct {
+// KVs interface
+type KVs interface {
+	GetValueOr(key interface{}, defValue interface{}) interface{}
+	Contains(key interface{}) bool
+	IfContains(key interface{}, action func(value interface{})) KVs
+	Put(key interface{}, value interface{}) KVs
```

Comment:
I think this KVs should be immutable, so I don't want to expose any method that could modify Kvs